core/src/test/java/kafka/admin/AclCommandTest.java

```diff
@@ -0,0 +1,589 @@
+                        list.addAll(commands);
+                        return list;
+                    });
+
+            List<String> args = new ArrayList<>(cmdArgs);
+            args.addAll(getCmd(ALLOW));
+            args.addAll(resourceCommand);
+            args.addAll(cmd);
+            args.add(ADD);
+            callMain(args);
+
+            for (Map.Entry<Set<ResourcePattern>, Set<AccessControlEntry>> resourcesToAclsEntry : resourcesToAcls.entrySet()) {
+                for (ResourcePattern resource : resourcesToAclsEntry.getKey()) {
+                    withAuthorizer(authorizer -> TestUtils.waitAndVerifyAcls(asScalaSet(resourcesToAclsEntry.getValue()), authorizer, resource, ANY));
+                }
+            }
+            List<String> resourceCmd = new ArrayList<>(resourceCommand);
+            resourceCmd.addAll(cmd);
+            testRemove(cmdArgs, resourcesToAcls.keySet().stream().flatMap(Set::stream).collect(Collectors.toSet()), resourceCmd);
+        }
+    }
+
+    private void testAclsOnPrefixedResources(List<String> cmdArgs) {
+        List<String> cmd = Arrays.asList("--allow-principal", PRINCIPAL.toString(), "--producer", "--topic", "Test-", "--resource-pattern-type", "Prefixed");
+
+        List<String> args = new ArrayList<>(cmdArgs);
+        args.addAll(cmd);
+        args.add(ADD);
+        callMain(args);
+
+        withAuthorizer(authorizer -> {
+            AccessControlEntry writeAcl = new AccessControlEntry(PRINCIPAL.toString(), WILDCARD_HOST, WRITE, ALLOW);
+            AccessControlEntry describeAcl = new AccessControlEntry(PRINCIPAL.toString(), WILDCARD_HOST, DESCRIBE, ALLOW);
+            AccessControlEntry createAcl = new AccessControlEntry(PRINCIPAL.toString(), WILDCARD_HOST, CREATE, ALLOW);
+            TestUtils.waitAndVerifyAcls(
+                    asScalaSet(new HashSet<>(Arrays.asList(writeAcl, describeAcl, createAcl))),
+                    authorizer,
+                    new ResourcePattern(TOPIC, "Test-", PREFIXED),
+                    ANY
+            );
+        });
+
+        args = new ArrayList<>(cmdArgs);
+        args.addAll(cmd);
+        args.add("--remove");
+        args.add("--force");
+        callMain(args);
+
+        withAuthorizer(authorizer -> {
+            TestUtils.waitAndVerifyAcls(
+                    asScalaSet(Collections.emptySet()),
+                    authorizer,
+                    new ResourcePattern(CLUSTER, "kafka-cluster", LITERAL),
+                    ANY
+            );
+            TestUtils.waitAndVerifyAcls(
+                    asScalaSet(Collections.emptySet()),
+                    authorizer,
+                    new ResourcePattern(TOPIC, "Test-", PREFIXED),
+                    ANY
+            );
+        });
+    }
+
+    private static Map<Set<ResourcePattern>, Set<AccessControlEntry>> producerResourceToAcls(boolean enableIdempotence) {
+        return new HashMap<Set<ResourcePattern>, Set<AccessControlEntry>>() {{
```

Comment:
Could you please use following code to replace it?
```java
    private static Map<Set<ResourcePattern>, Set<AccessControlEntry>> producerResourceToAcls(boolean enableIdempotence) {
        Map<Set<ResourcePattern>, Set<AccessControlEntry>> result = new HashMap<>();
        result.put(TOPIC_RESOURCES, asJavaSet(AclCommand.getAcls(asScalaSet(USERS), ALLOW, asScalaSet(
                new HashSet<>(Arrays.asList(WRITE, DESCRIBE, CREATE))), asScalaSet(HOSTS))));
        result.put(TRANSACTIONAL_ID_RESOURCES, asJavaSet(AclCommand.getAcls(asScalaSet(USERS), ALLOW, asScalaSet(
                new HashSet<>(Arrays.asList(WRITE, DESCRIBE))), asScalaSet(HOSTS))));
        result.put(Collections.singleton(CLUSTER_RESOURCE), asJavaSet(AclCommand.getAcls(asScalaSet(USERS), ALLOW,
                enableIdempotence ? asScalaSet(Collections.singleton(IDEMPOTENT_WRITE)) : asScalaSet(Collections.emptySet()), asScalaSet(HOSTS))));
        return result;
    }
```

Reply:
Done